src/liblzma/meson.build

```diff
@@ -0,0 +1,358 @@
+        ]
+    endif
+endif
+
+if get_option('decoders').length() > 0
+    lzma_src += [
+        'common/alone_decoder.c',
+        'common/auto_decoder.c',
+        'common/block_buffer_decoder.c',
+        'common/block_decoder.c',
+        'common/block_header_decoder.c',
+        'common/easy_decoder_memusage.c',
+        'common/file_info.c',
+        'common/filter_buffer_decoder.c',
+        'common/filter_decoder.c',
+        'common/filter_flags_decoder.c',
+        'common/index_decoder.c',
+        'common/index_hash.c',
+        'common/stream_buffer_decoder.c',
+        'common/stream_decoder.c',
+        'common/stream_flags_decoder.c',
+        'common/vli_decoder.c'
+    ]
+
+    if have_threads == true
+        lzma_src += [
+            'common/stream_decoder_mt.c',
+        ]
+    endif
+
+    if get_option('microlzma') == true
+        lzma_src += [
+            'common/microlzma_decoder.c'
+        ]
+    endif
+
+    if have_lzip == true
+        lzma_src += 'common/lzip_decoder.c'
+    endif
+endif
+
+# check/Makefile.inc
+
+lzma_src += [
+    'check/check.c'
+]
+
+if get_option('small') == true
+    lzma_src += [
+        'check/crc32_small.c'
+    ]
+else
+    lzma_src += [
+        'check/crc32_table.c'
+    ]
+
+    if have_assembler_x86 == true
+        lzma_src += [
+            'check/crc32_x86.S'
+        ]
+    else
+        lzma_src += [
+            'check/crc32_fast.c'
+        ]
+    endif
+endif
+
+if get_option('checks').contains('crc64') == true
+    if get_option('small') == true
+        lzma_src += [
+            'check/crc64_small.c'
+        ]
+    else
+        lzma_src += [
+            'check/crc64_table.c'
+        ]
+
+        if have_assembler_x86 == true
+            lzma_src += [
+                'check/crc64_x86.S'
+            ]
+        else
+            lzma_src += [
+                'check/crc64_fast.c'
+            ]
+        endif
+    endif
+endif
+
+if get_option('checks').contains('sha256') == true and get_option('external-sha256') == false
+    lzma_src += [
+        'check/sha256.c'
+    ]
+endif
+
+if get_option('encoders').contains('lzma1') == true or
+   get_option('decoders').contains('lzma1') == true or
+   get_option('encoders').contains('lzma2') == true or
+   get_option('decoders').contains('lzma2') == true
+
+    # lz/Makefile.inc
+    if get_option('encoders').contains('lzma1') == true or
+       get_option('encoders').contains('lzma2') == true
+        lzma_src += [
+            'lz/lz_encoder.c',
+            'lz/lz_encoder_mf.c'
+        ]
+    endif
+
+    if get_option('decoders').contains('lzma1') == true or
+       get_option('decoders').contains('lzma2') == true
+        lzma_src += [
+            'lz/lz_decoder.c'
+        ]
+    endif
+endif
+
+if get_option('encoders').contains('lzma1') == true or
+   get_option('decoders').contains('lzma1') == true
+
+    # lzma/Makefile.inc
+    lzma_src += [
+        'lzma/lzma_encoder_presets.c'
+    ]
```

Comment:
Only this line needs to be inside this `if`, the rest of the lines can be outside since their comparisons are a subset of this `if`.

Reply:
not so simple: lzma/Makefile.inc is included if lzma1 is enabled, but in lzma/Makefile.inc there are some code if lzma2 is enabled